src/common/cups_print.c

```diff
@@ -313,6 +333,61 @@ GList *dt_get_papers(const char *printer_name)
   return result;
 }
 
+GList *dt_get_media_type(const dt_printer_info_t printer)
+{
+  const char *printer_name = printer.name;
+  GList *result = NULL;
+
+  // check now PPD media type
+
+  const char *PPDFile = cupsGetPPD(printer_name);
```

Comment:
A general remark: The PPD api in CUPS is deprecated. Maybe we should switch to the newer one eventually?

https://www.cups.org/doc/api-ppd.html

Reply:
Yes, I had this on my todo list for some time now. I plan to propose a patch in this direction for 2.6.